sycl/include/CL/sycl/device_selector.hpp

```diff
@@ -27,26 +33,56 @@ class __SYCL_EXPORT device_selector {
   virtual int operator()(const device &device) const = 0;
 };
 
+/// The default selector chooses the first available SYCL device. Devices of
+/// different kinds are prioritized in the following order:
+/// 1. GPU
+/// 2. Accelerator
+/// 3. CPU
+/// 4. Host
```

Comment:
One caveat in adding such comments is that they must be updated together with the implementation.
As implementation is located in other file, it will be very hard to keep them in sync.

Reply:
@bader makes sense. I can move part of this comment to .cpp file, it will still be rendered in doxygen.